xfbin/structure/br/br_nut.py

```diff
@@ -4,64 +4,152 @@
 # Based on Smash Forge Nut implementation
 # https://github.com/jam1garner/Smash-Forge/blob/master/Smash%20Forge/Filetypes/Textures/NUT.cs
 class BrNut(BrStruct):
-    def __br_read__(self, br: BinaryReader) -> None:
+    def __br_read__(self, br: BinaryReader):
         self.magic = br.read_str(4)
+        #print(self.magic)
 
         if self.magic != "NTP3":
-            raise Exception('Invalid NUD magic.')
+            raise Exception('Invalid NUT magic.')
 
         self.version = br.read_uint16()
 
-        self.textureCount = br.read_uint16()
+        self.texture_count = br.read_uint16()
 
         br.read_uint32(2)
+        #print(len(br.buffer()))
+        self.textures = br.read_struct(BrNutTexture, self.texture_count, self)
+    
+    def __br_write__(self, br: 'BinaryReader', nut: 'Nut'):
+        br.write_str('NTP3')
+        br.write_uint16(0x0100)
+        br.write_uint16(len(nut.textures))
+        
+        br.write_uint64(0)
+
+        for tex in nut.textures:
+            br.write_struct(BrNutTexture(), tex)
 
-        self.textures = br.read_struct(BrNutTexture, self.textureCount, self)
 
 
 class BrNutTexture(BrStruct):
-    def __br_read__(self, br: BinaryReader, nut: BrNut) -> None:
-        self.totalSize = br.read_uint32()
+    def __br_read__(self, br: BinaryReader, nut: BrNut):
+        self.total_size = br.read_uint32()
         br.read_uint32()
 
-        self.dataSize = br.read_uint32()
-        self.headerSize = br.read_uint16()
+
+        self.data_size = br.read_uint32()
+        self.header_size = br.read_uint16()
         br.read_uint16()
 
         br.read_uint8()
-        self.mipmapCount = br.read_uint8()
+        self.mipmap_count = br.read_uint8()
         br.read_uint8()
-        self.pixelFormat = br.read_uint8()
+        self.pixel_format = br.read_uint8()
 
         self.width = br.read_uint16()
         self.height = br.read_uint16()
 
         br.read_uint32()
-        self.cubeMapFormat = br.read_uint32()
+        self.cubemap_format = br.read_uint32()
 
-        self.isCubemap = False
-        if self.cubeMapFormat & 0x200:
-            self.isCubemap = True
+        self.is_cube_map = False
+        if self.cubemap_format & 0x200:
+            self.is_cube_map = True
+        #print(self.is_cube_map)
```

Comment:
The commented out print calls in this function (and the writer function) should be removed